### PR TITLE
Predict endpoint

### PR DESCRIPTION
**Pull request by morten-j**
yo

### CODE REVIEW

File: sanic/src/utils.py [hunk 1]
```diff
@@ -129,4 +129,11 @@ def build_feature_extractor():
...
-    return keras.Model(inputs, outputs, name="feature_extractor")
\ No newline at end of file
+    return keras.Model(inputs, outputs, name="feature_extractor")
+
+
+def load_model(path):
+    if os.path.exists(path):
+        return keras.models.load_model(path)
+    else:
+        return "Blyat"
```
Comment (morten-j) [inline sanic/src/utils.py]: Stop troll

Comment (CasperKjaerhus) [reply sanic/src/utils.py]: ja man wth

Comment (CasperKjaerhus) [inline sanic/src/utils.py]:
```suggestion
        return "this is cringe" 
```

File: sanic/src/server.py [hunk 2]
```diff
@@ -3,9 +3,16 @@
...
 from sanic_ext import openapi
 from pathlib import Path
+import numpy as np
 import feature_extraction as fe
 import tempfile
 import os
+import utils
+
+model = utils.load_model(os.path.join("models", "test")) #TODO fix whhen there is a model
+feature_extractor = utils.build_feature_extractor() #TODO Fjern hvis den ikke skal bruges
```
Comment (CasperKjaerhus) [inline sanic/src/server.py:13]: Er det muligt at bruge en model flere gange concurrently? for ellers kan det blive et problem med flere requests

File: sanic/src/server.py [hunk 3]
```diff
@@ -93,4 +100,55 @@ async def savevideo(request: Request) -> HTTPResponse:
         file.write(videofile.body)
         file.close()
     else:
-        return text("No label provided!", 400)
\ No newline at end of file
+        return text("No label provided!", 400)
```
Comment (CasperKjaerhus) [inline sanic/src/server.py:103]: cringe at du stjæler min linje

File: sanic/src/server.py [hunk 4]
```diff
@@ -93,4 +100,55 @@ async def savevideo(request: Request) -> HTTPResponse:
...
+    """
+    Predict sign from video
+    openapi:
+    operationId: predict_video
+    tags:
+      - Predict sign
+    parameters:
+      - name: label
+        in: query
+        type: string
```
Comment (CasperKjaerhus) [inline sanic/src/server.py]: skal man bruge en label til at predict?

File: sanic/src/server.py [hunk 5]
```diff
@@ -93,4 +100,55 @@ async def savevideo(request: Request) -> HTTPResponse:
...
+    # Expand dims size the model expects a list of videos and not just a video
+    fixed_size = np.expand_dims(video, axis=0)
+    prediction = model.predict(fixed_size)
+
+    # Find index of the max value
+    max_value_index = np.argmax(prediction)
+
+    # Create object to be returned
+    returnObject = dict()
+    returnObject["label"] = label
```
Comment (CasperKjaerhus) [inline sanic/src/server.py]: Clienten burde jo vide hvilket label den har sendt allerede?

File: sanic/src/server.py (hunk 3 above)
Comment (morten-j) [reply sanic/src/server.py:103]: - [x] Beta

**Review by CasperKjaerhus — APPROVED** [review]
Fix at label bliver sendt og så er den nice <3
god kode <3

File: sanic/src/server.py (hunk 3 above)
Comment (mKiddeT) [reply sanic/src/server.py:103]: `if (CasperKjaerhus == Cringe) {
    print ("He is cringe")
}`
OUTPUT:
He is cringe

**Review by mKiddeT — APPROVED** [review]
Det Cleth sagde
